bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java

```diff
@@ -0,0 +1,185 @@
+import com.adobe.cq.wcm.core.components.models.contentfragment.ContentFragmentList;
+import com.adobe.cq.wcm.core.components.models.contentfragment.DAMContentFragment;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.day.cq.search.PredicateGroup;
+import com.day.cq.search.Query;
+import com.day.cq.search.QueryBuilder;
+import com.day.cq.search.result.SearchResult;
+import com.day.cq.tagging.TagConstants;
+
+@Model(
+        adaptables = SlingHttpServletRequest.class,
+        adapters = {
+                ContentFragmentList.class,
+                ComponentExporter.class
+        },
+        resourceType = ContentFragmentListImpl.RESOURCE_TYPE
+)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class ContentFragmentListImpl implements ContentFragmentList {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ContentFragmentListImpl.class);
+
+    public static final String RESOURCE_TYPE = "core/wcm/components/contentfragmentlist/v1/contentfragmentlist";
+
+    public static final String DEFAULT_DAM_PARENT_PATH = "/content/dam";
+
+    @Self(injectionStrategy = InjectionStrategy.REQUIRED)
+    private SlingHttpServletRequest slingHttpServletRequest;
+
+    @Inject
+    private ContentTypeConverter contentTypeConverter;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    @ValueMapValue(name = ContentFragmentList.PN_MODEL_PATH, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String modelPath;
+
+    @ValueMapValue(name = ContentFragmentList.PN_ELEMENT_NAMES, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String[] elementNames;
+
+    @ValueMapValue(name = ContentFragmentList.PN_TAG_NAME, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String[] tagNames;
+
+    @ValueMapValue(name = ContentFragmentList.PN_FOLDER_NAME, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String parentPath;
+
+    private List<DAMContentFragment> items = new ArrayList<>();
+
+    @PostConstruct
+    private void initModel() {
+        // Default path limits search to DAM
+        if (StringUtils.isEmpty(parentPath)) {
+            parentPath = DEFAULT_DAM_PARENT_PATH;
+        }
+
+        if (StringUtils.isEmpty(modelPath)) {
+            LOG.warn("Please provide a model path");
+            return;
+        }
+
+        Session session = resourceResolver.adaptTo(Session.class);
+        if (session == null) {
+            LOG.warn("Session was null therefore no query was executed");
+            return;
+        }
+
+        QueryBuilder queryBuilder = resourceResolver.adaptTo(QueryBuilder.class);
+        if (queryBuilder == null) {
+            LOG.warn("Query builder was null therefore no query was executed");
+            return;
+        }
+
+        Map<String, String> queryParameterMap = new HashMap<>();
+        queryParameterMap.put("path", parentPath);
```

Comment:
Do we need to add p.limit = -1 to the queryParameterMap?

Reply:
Could you please elaborate?